src/message/contentTypes/textMessage.js

```diff
@@ -10,7 +10,13 @@ export default class TextMessage extends Message {
 
     setText(text) {
         validate.stringLength(text, null, 640, 'text', 'TextMessage.setText');
-        this.text = text;
+        this.state.text = text;
+        return this;
+    }
+
+    withTag(tag) {
+        validate.oneOf(tag, ['ISSUE_RESOLUTION'], 'tag', 'TextMessage.withTag');
+        this.tag = tag;
```

Comment:
`this.state` is the message and `tag` lives outside of the message which is why it's on `this.tag`